### ISSUE TITLE
Taildrive shares (and maybe host sharing too?) seem borked on Windows

### ISSUE THREAD

**Issue by FlorentLM** [tailscale/tailscale#13912]
### What is the issue?

I have a Linux machine and a Windows machine both configured to share folders through Taildrive. The two devices are tagged, but the same problem happens on a Windows host under my user.

`tailscale drive list` shows the shared folders on both machine, but the Windows one has nothing in the `as` field, whereas the Linux one has my local user.

I can access the folder shared by the Linux host, not the one shared by the Windows host (it actually makes my WebDAV client crash when I try to access it).

Yesterday I also tried sharing the Windows machine to a coworker (in hopes for them to access the files via classic SMB) but they could not access it either.

Has anything changed recently? I used to be able to access both hosts via Taildrive _and_ both hosts via classic SMB (using tailscale serve)

Note that I can't access the Windows shared drive from any other host (macOS, Linux, Android, etc)

### Steps to reproduce

Linux host:
`tailscale drive share data /path/to/data`
`tailscale drive list`

Windows host:
`tailscale drive share data D:\`
`tailscale drive list`

Try to access the shares from a third host by mounting `http:/100.100.100.100:8080` - The Linux one works, the Windows one does not

### OS

Linux, macOS, Windows

### OS version

Debian 11, Windows 11

### Tailscale version

1.76.1 on Linux, 1.76.3 on Windows

**Comment by linux4life798** [comment]
I've seen a similar directory listing issue on Debian, when there was a special file in the directory being shared #13631. It might be reasonable to guess that there are other file situation and edge cases that aren't being handled correctly (maybe odd permissions...)

I know this isn't a solution, but, for the sake of debugging, have you tried unsharing all Windows drives and then sharing a single empty directory? See if you can access this simple share from a different device, without erroneous behavior on the client side. Then, maybe try to add one file to the Windows share from a remote machine.

Also, beware, I consistently see errors loading random files #13632.

**Comment by FlorentLM** [comment]
Yes, I've tried only sharind an empty folder, on both the C: drive and another drive, but this changes nothing unfortunately. I see the Windows machines in `http:/100.100.100.100:8080/xx@gmail.com` but no shares visible under them. Trying to create a folder or a file in the machine root directory raises an error (as expected since it's not in a share)

**Comment by Nerodacles** [comment]
I’m experiencing the same problem. When I create a shared folder on Windows, it doesn’t show up on other machines, whether they’re running Windows or Linux. However, when I create a shared folder on Linux, it’s accessible from all devices without any issues.

**Comment by kasperisager** [comment]
I hit this myself today on Tailscale 1.80.2; drives shared from my Linux machines are showing up, but not those shared from my Windows machines.

**Comment by casperfrx** [comment]
Just reporting that I'm having this issue a well. A Webdav Android client repots a HTTP 502 Bad Gateway error for what it's worth when going to 100.100.100.100:8080/[windows-machine-name]/. Ofc the Android Webdav client works fine with drives shared from Linux.

**Comment by Mansarde** [comment]
Same thing for me on v1.84.0 on Windows (24H2).
I can see my tailnet domain as a folder in the WebDAV folder, but there's nothing inside, no matter what path I share.

**Comment by oddlid** [comment]
I just tried to share both from macOS and Linux and access the shares from the files app in iPadOS/iOS. The Linux share works fine, but for the mac shares, I can see the top level share names, but if I try to enter any of them, I just get a spinner saying ”reading…”, which never stops until I go back up one level.

**Comment by linux4life798** [comment]
There was an [issue handling directories with spaces](https://github.com/tailscale/tailscale/issues/13632) which is now fixed in `v1.88.1`. It might be worth retrying, once your client tailscale (on the accessing/client/browsing side -- not the file share/server side) updates.

**Comment by khanhtd36arb** [comment]
We’re seeing this on Windows as well, and the behavior matches the bug report closely.

Environment:
- Tailscale client: 1.96.3-t3ffddb134-g460d8764a on Windows
- Node: `arbin-vn-khanhtd`
- Tailnet: `arbin-vndev.github`
- Local ACL grants `drive:share` and `drive:access` via `nodeAttrs` for `autogroup:member`

What I observed:
- `tailscale drive list` on the Windows host shows the shares as configured:
  - `workspace    E:\`
  - `tdtest       E:\MITS11\taildrive-test`
- `tailscale status --json` on the normal user session shows Taildrive capabilities present (`drive:share`, `drive:access`).
- The Taildrive WebDAV server responds at `http://100.100.100.100:8080/`.
- The tailnet root is visible with `PROPFIND /` and `PROPFIND /arbin-vndev.github/`.
- However, the Windows machine folder does not expose the actual share paths:
  - `PROPFIND http://100.100.100.100:8080/arbin-vndev.github/arbin-vn-khanhtd/` => `404 Not Found`
  - `PROPFIND http://100.100.100.100:8080/arbin-vndev.github/arbin-vn-khanhtd/workspace/` => `404 Not Found`
- Re-sharing the folder (`tailscale drive unshare workspace` then `tailscale drive share workspace E:\`) did not change the WebDAV tree.
- `gsudo tailscale drive list` returns `Access denied: taildrive sharing not enabled, please add the attribute "drive:share" to this node in your ACLs' "nodeAttrs" section`, but the non-elevated session already has the capability map showing `drive:share` and `drive:access`.

Relevant local evidence:
- `tailscale drive list` output
- `tailscale status --json` capability map
- `curl.exe -X PROPFIND` results against the Taildrive WebDAV server
- `tailscale bugreport` id: `BUG-74f64582a56fa37a4350dda573dad5b30454bf6f6a93a72574395598e1dd1e9f-20260417040435Z-7570ee0ebfa26107`

This looks like the Windows Taildrive publication bug where shares are configured locally but not actually exposed in the WebDAV tree.
